### PR TITLE
CD-171: XClarity (Lenovo) Storage Health

### PR DESCRIPTION
**Pull request by mariemm97**

### CODE REVIEW

File: examples/ssh/xclarity_storage_health.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        ]
+    },
+    "tags": [
+        "xclarity",
+        "lenovo",
+        "storage",
+        "health",
+        "status"
+    ],
+    "sample_period": 1800,
```
Comment (kgskgs) [inline examples/ssh/xclarity_storage_health.json:22]: change to sample_period_s

File: examples/ssh/xclarity_storage_health.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "version": "1.0.0",
+    "name": "XClarity Storage Health",
+    "description": "Monitors the health status of storage systems on Lenovo XClarity server",
+    "category": "Software Integration",
+    "requirements": {
+        "credentials": true,
+        "sandbox_version": "1.12",
+        "others": [
+            "SSH Enabled",
+            "The keyboard_interactive option can be true or false depending on the XClarity version"
```
Comment (kgskgs) [inline examples/ssh/xclarity_storage_health.json:12]: this doesn't really say anything

**Review by kgskgs — APPROVED** [review]

File: examples/ssh/xclarity_storage_health.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ */
+function validate() {
+    executeCommand(command)
+        .then(parseValidateOutput)
+        .then(D.success)
+        .catch(checkSshError);
+}
+
+function parseValidateOutput(output) {
+    if (output.trim() !== "") {
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_storage_health.js:58]: please validate properly, this does not check if the command exists or not

File: examples/ssh/xclarity_storage_health.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+// Parse and insert storage health data into the custom driver variables
+function parseData(output) {
+    var cleanedOutput = output.replace(/Flash DIMMs[\s\S]*$/, '');
+    var lines = cleanedOutput.split("\n");
+    var drivesData = [];
+    for (var i = 3; i < lines.length-2; i++) {
+        var line = lines[i].trim();
+        var parts = line.split(/\s+/);
+        var fruName = parts.slice(0, -1).join(" "); // Join all parts except the last one as FRU Name
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_storage_health.js:82]: Maybe use a more descriptive name or at least comment
(I guess this is the field replaceable unit?)

File: examples/ssh/xclarity_storage_health.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+    "logo": "lenovo-xclarity.svg",
+    "version": "1.0.0",
+    "name": "XClarity Storage Health",
+    "description": "Monitors the health status of storage systems on Lenovo XClarity server",
+    "category": "Software Integration",
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_storage_health.json:6]: Maybe hardware integration

File: examples/ssh/xclarity_storage_health.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        ]
+    },
+    "tags": [
+        "xclarity",
+        "lenovo",
+        "storage",
+        "health",
+        "status"
+    ],
+    "sample_period_s": 1800,
```
Comment (GeorgiTsatsev) [inline examples/ssh/xclarity_storage_health.json:21]: Bump to 30 minutes

**Review by GeorgiTsatsev — APPROVED** [review]

**Pull request merged**